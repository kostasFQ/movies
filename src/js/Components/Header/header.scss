@import url('https://fonts.googleapis.com/css?family=Fahkwang');

.container {
  color: white;
  display: flex;
  flex-direction: column;
  background-image: url('../../../imgs/cover-image.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  height: 400px;
}

.inner {
  background-image: linear-gradient(rgba(0, 0, 0, 0.9) 0%, rgba(0, 0, 0, 0) 100%);
  padding: 10px 50px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-family: 'Fahkwang', sans-serif;
}

input[type=text] {
  border: 2px solid white;
  border-radius: 4px;
  background: none;
  height: 2em;
  color: white;

  &::placeholder {
    color: white;
  }
}
