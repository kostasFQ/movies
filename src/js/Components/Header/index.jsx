import React from 'react';
import s from './header.scss'

const Header = () => {
  return (
    <div className={s.container}>
      <div className={s.inner}>
        <h1>FILMS</h1>
        <div>
          <input className={s.input} type="text" placeholder='search'/>
        </div>
      </div>
    </div>
  );
};
export default Header;