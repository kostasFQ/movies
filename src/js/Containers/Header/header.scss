.container {
  height: 150px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-family: 'Fahkwang', sans-serif;
  position: absolute;
  width: 100%;
}

.title {
  display: flex;
  color: white;
  margin-left: 3%;
}

input[type=text] {
  margin-top: 20px;
  border: 2px solid white;
  border-radius: 4px;
  background: none;
  height: 2em;
  width: 90%;
  color: white;
  background-image: url('../../../imgs/icons/search-3-32.png');
  background-size: 16px;
  background-repeat: no-repeat;
  background-position: 90%;

  &::placeholder {
    color: white;
  }

  @media only screen and (max-width: 600px) {
    width: 60%;
  }
}
