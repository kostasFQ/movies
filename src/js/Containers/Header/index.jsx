import React from 'react';
import PropTypes from 'prop-types';
import s from './header.scss';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      str: '',
    };
  }

  toState = (e) => {
    this.setState({ str: e.target.value });
  }

  send = (e) => {
    const { getMovie } = this.props;
    const { str } = this.state;
    e.preventDefault();
    getMovie(str);
    this.setState({ str: '' });
  }

  render() {
    const { str } = this.state;
    return (
      <div className={s.container}>
        <h1 className={s.title}>FILMS</h1>
        <form onSubmit={this.send}>
          <input className={s.input} value={str} type="text" placeholder="search" onChange={this.toState} />
        </form>
      </div>
    );
  }
}
export default Header;

Header.propTypes = {
  getMovie: PropTypes.func.isRequired,
};
