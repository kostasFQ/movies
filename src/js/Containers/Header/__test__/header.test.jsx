import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { movie } from '../../../../../configs/jest/__mocks__/mockStore';
import Header from '../index';

const mockStore = configureMockStore([thunk]);

const initialState = { movie };
const store = mockStore(initialState);

test('1.test HEADER component render', () => {
  const testRenderer = TestRenderer.create(
    <Provider store={store}><Header /></Provider>,
  );
  const result = testRenderer.toJSON();

  expect(result.type).toBe('div');
  expect(result).toMatchSnapshot();
});

test('2.test HEADER component search', () => {
  const e = { preventDefault: () => 'x' };
  const getMovie = film => film;
  const testRenderer = TestRenderer.create(
    <Provider store={store}><Header getMovie={getMovie} /></Provider>,
  );
  const testInstance = testRenderer.root;

  const form = testInstance.findByType('form');
  form.props.onSubmit(e);
});

test('3.test HEADER component input', () => {
  const e = { target: { value: 'x' } };
  const testRenderer = TestRenderer.create(
    <Provider store={store}><Header /></Provider>,
  );
  const testInstance = testRenderer.root;

  const inp = testInstance.findByType('input');
  inp.props.onChange(e);
});
