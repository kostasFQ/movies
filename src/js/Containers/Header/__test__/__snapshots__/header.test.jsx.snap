// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.test HEADER component render 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    FILMS
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="search"
      type="text"
      value=""
    />
  </form>
</div>
`;
